src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Net.Http;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Executors;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.AuthScheme.PoP;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+using Microsoft.Identity.Client.TelemetryCore.Internal.Events;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Parameter builder for the <see cref="IPublicClientApplication.AcquireTokenByUsernamePassword(IEnumerable{string}, string, string)"/>
+    /// operation. See https://aka.ms/msal-net-up
+    /// </summary>
+    public sealed class AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder :
```

Comment:
The issue recommends against the moving of the current api to the base class and creating a new one. Not entirely sure this is the best way unless there is some reason you were thinking of? Perhaps I misunderstood what you mean exactly? Thoughts on this @bgavrilMS

Reply:
Well we have AcquireTokenSilent on the base class and it's a struggle to reason about it - we keep having to check `IsConfidentialClient` etc.

I'm thinking more for the future PCA / CCA split.